File: python/app/domain/user_unique_checker.py

from abc import ABC, abstractmethod

class UserUniqueChecker(ABC):
    """用户唯一性检查接口"""

    @abstractmethod
    async def is_username_unique(self, username: str) -> bool:
        pass

    @abstractmethod
    async def is_nickname_unique(self, nickname: str) -> bool:
        pass

    @abstractmethod
    async def is_mobile_unique(self, mobile: str) -> bool:
        pass

File: dotnet/src/domain/UserUniqueChecker.cs
namespace DddExample.Domain
{
    /// <summary>
    /// 用户唯一性检查接口
    /// </summary>
    public interface UserUniqueChecker
    {
        /// <summary>
        /// 检查用户名是否唯一
        /// </summary>
        bool IsUsernameUnique(string username);

        /// <summary>
        /// 检查昵称是否唯一
        /// </summary>
        bool IsNicknameUnique(string nickname);

        /// <summary>
        /// 检查手机号是否唯一
        /// </summary>
        bool IsMobileUnique(string mobile);
    }
}
